src/plugins/ssu_catch.rs

```diff
@@ -197,53 +197,40 @@ impl SsufidPlugin for SsuCatchPlugin {
 
     async fn crawl(&self, posts_limit: u32) -> Result<Vec<SsufidPost>, PluginError> {
         let pages = posts_limit / Self::POSTS_PER_PAGE + 1;
-        let mut page_futures = Vec::new();
-
-        for page in 1..=pages {
-            page_futures.push(self.fetch_page_posts(page));
-        }
 
         // 모든 페이지 크롤링이 완료될 때까지 대기
-        let page_results = futures::future::join_all(page_futures).await;
-        let mut all_metadata = Vec::new();
-
-        for page_result in page_results {
-            let metadata_items = page_result?;
-            for metadata in metadata_items {
-                all_metadata.push(metadata);
-                // posts_limit에 도달하면 더 이상 메타데이터 추가하지 않음
-                if all_metadata.len() >= posts_limit as usize {
-                    break;
-                }
-            }
-        }
-
-        let mut content_futures = Vec::new();
+        let metadata_results =
+            futures::future::join_all((1..=pages).map(|page| self.fetch_page_posts(page))).await;
 
-        for metadata in &all_metadata {
-            content_futures.push(self.fetch_post_content(&metadata.url));
-        }
+        let all_metadata = metadata_results
+            .into_iter()
+            .filter_map(|result| result.ok())
+            .flatten()
+            .take(posts_limit as usize)
+            .collect::<Vec<SsuCatchMetadata>>();
```

Comment:
만약 하나라도 실패해도 오류를 리턴하게 하고 싶다면 이렇게 할 수 있어요:
```suggestion
            .take(posts_limit as usize)
            .collect::<Result<Vec<SsuCatchMetadata>, Error>>();
```